acceptance/tests/utf8/utf8-in-puppet-describe.rb

```diff
@@ -0,0 +1,71 @@
+test_name 'utf-8 characters in module doc string, puppet describe' do
+  # utf8chars = "€‰ㄘ万竹ÜÖ"
+  utf8chars = "\u20ac\u2030\u3118\u4e07\u7af9\u00dc\u00d6"
+  agents.each do |agent|
+    mod_dir = agent.tmpdir("describe")
+
+    manifest =
+<<PP
+
+File {
+  ensure => directory,
+  mode => "0755",
+}
+
+file {
+  '#{mod_dir}/environments':;
+  '#{mod_dir}/environments/production':;
+  '#{mod_dir}/environments/production/modules':;
+  '#{mod_dir}/environments/production/modules/describe_module':;
+  '#{mod_dir}/environments/production/modules/describe_module/lib':;
+  '#{mod_dir}/environments/production/modules/describe_module/lib/puppet':;
+  '#{mod_dir}/environments/production/modules/describe_module/lib/puppet/type':;
+}
+
+file { '#{mod_dir}/environments/production/modules/describe_module/lib/puppet/type/describe.rb' :
+  ensure => file,
+  mode => '0640',
+  content => '
+Puppet::Type.newtype(:describe) do
+  @doc = "Testing to see if puppet handle describe blocks correctly
+when they contain utf8 characters, such as #{utf8chars}
+"
+end
+',
+}
+
+PP
+
+    step "Apply manifest" do
+      apply_manifest_on(
+        agent,
+        manifest,
+        {
+          :acceptable_exit_codes => [0, 2],
+          :catch_failures => true,
+          :enviornment => {:LANG => "en_US.UTF-8"}
+        }
+      )
+    end
+
+    step "puppet describe" do
+      result = on(
+        agent,
+        puppet(
+          "describe",
+          "describe",
+          "--environmentpath #{mod_dir}/environments"
+        ),
+        :enviornment => {:LANG => "en_US.UTF-8"}
+      )
+      assert_equal(result.exit_code, 0, "puppet describe failed.")
```

Comment:
true.  i hand-assert on exit codes to better aid debugging with custom assertion failure messaging and use `:accept_all_exit_codes => true`.  this also forces beaker errors to become failures.